solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/batch/SolrSearchSink.java

```diff
@@ -0,0 +1,154 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.plugin.batch;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.batch.Output;
+import co.cask.cdap.api.data.batch.OutputFormatProvider;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.BatchRuntimeContext;
+import co.cask.cdap.etl.api.batch.BatchSink;
+import co.cask.cdap.etl.api.batch.BatchSinkContext;
+import co.cask.cdap.format.StructuredRecordStringConverter;
+import co.cask.hydrator.plugin.common.SolrOutputFormat;
+import co.cask.hydrator.plugin.common.SolrRecordWriter;
+import co.cask.hydrator.plugin.common.SolrSearchSinkConfig;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.apache.hadoop.io.Text;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Batch SolrSearch Sink Plugin - Writes data to a SingleNode Solr or to SolrCloud.
+ */
+@Plugin(type = BatchSink.PLUGIN_TYPE)
+@Name("SolrSearch")
+@Description("This plugin allows users to build the pipelines to write data to Solr. The input fields coming from " +
+  "the previous stage of the pipeline are mapped to Solr fields. User can also specify the mode of the Solr to " +
+  "connect to. For example, SingleNode Solr or SolrCloud.")
+public class SolrSearchSink extends BatchSink<StructuredRecord, Text, Text> {
+  private static final Gson GSON = new Gson();
+  private static final Type SCHEMA_TYPE = new TypeToken<Schema>() { }.getType();
+  private final BatchSolrSearchConfig batchConfig;
+
+  public SolrSearchSink(BatchSolrSearchConfig batchConfig) {
+    this.batchConfig = batchConfig;
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext context) throws Exception {
+    context.addOutput(Output.of(batchConfig.referenceName, new SolrSearchSink.SolrOutputFormatProvider(batchConfig)));
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    batchConfig.validateSolrConnectionString();
+    if (inputSchema != null) {
+      batchConfig.validateKeyField(inputSchema);
+      batchConfig.validateInputFieldsDataType(inputSchema);
+    }
+  }
+
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    batchConfig.testSolrConnection();
```

Comment:
As discussed and suggested by @albertshau , at configuration time, we should only be checking static things, which means things that will never change.
However, calling testConnection() inside transform method will not be feasible at all, as it will be called for each record.
So the last option that seems feasible (Also @anwar6953 has suggested the same), was to call testConnection() in initialize() method. And also, it would be good to check connection before each mapper, so that we can catch the failures atleast before transform method.
For discussion details, please refer:
https://github.com/caskdata/hydrator-plugins/pull/425/files/b1c95907c73cf27a2d0bf514a704ca8ed9d8d2dd#r77234560 
Added comments in the code also.

Please let me know, if this is OK?